aspnet/web-api/overview/advanced/calling-a-web-api-from-a-net-client.md

```diff
@@ -29,106 +29,141 @@ In this tutorial, a client app is written that consumes the following web API.
+```csharp
+var formatters = new List<MediaTypeFormatter>() {
+    new MyCustomFormatter(),
+    new JsonMediaTypeFormatter(),
+    new XmlMediaTypeFormatter()
+};
+resp.Content.ReadAsAsync<IEnumerable<Product>>(formatters);
+```
 
 For more information, see [Media Formatters in ASP.NET Web API 2](../formats-and-model-binding/media-formatters.md)
 
 ## Sending a POST Request to Create a Resource
 
 The following code sends a POST request that contains a `Product` instance in JSON format:
 
-[!code-csharp[Main](calling-a-web-api-from-a-net-client/samples/sample8.cs)]
+[!code-csharp[Main](calling-a-web-api-from-a-net-client/sample/client/Program.cs?name=snippet_CreateProductAsync)]
+
+The **PostAsJsonAsync** method:
 
-The **PostAsJsonAsync** method serializes an object to JSON and then sends the JSON payload in a POST request. If the request succeeds, it should return a 201 (Created) response, with the URL of the created resources in the Location header.
+* Serializes an object to JSON.
+* Sends the JSON payload in a POST request.
+
+If the request succeeds:
+
+* It should return a 201 (Created) response.
+* The response should include the URL of the created resources in the Location header.
 
 <a id="PuttingResource"></a>
 ## Sending a PUT Request to Update a Resource
 
-The following code sends a PUT request to update a product.
+The following code sends a PUT request to update a product:
 
-[!code-csharp[Main](calling-a-web-api-from-a-net-client/samples/sample9.cs)]
+[!code-csharp[Main](calling-a-web-api-from-a-net-client/sample/client/Program.cs?name=snippet_UpdateProductAsync)]
 
 The **PutAsJsonAsync** method works like **PostAsJsonAsync**, except that it sends a PUT request instead of POST.
 
 <a id="DeletingResource"></a>
 ## Sending a DELETE Request to Delete a Resource
 
-The following code sends a DELETE request to delete a product.
+The following code sends a DELETE request to delete a product:
+
+[!code-csharp[Main](calling-a-web-api-from-a-net-client/sample/client/Program.cs?name=snippet_DeleteProductAsync)]
 
-[!code-csharp[Main](calling-a-web-api-from-a-net-client/samples/sample10.cs)]
+Like GET, a DELETE request does not have a request body. You don't need to specify JSON or XML format with DELETE.
 
-Like GET, a DELETE request does not have a request body, so you don't need to specify JSON or XML format.
+## Testing the sample
 
-## Complete Code Example
+To test the client app:
 
-Here is the complete code for this tutorial. The code is very simple and doesn't include much error handling, but it shows the basic CRUD operations using **HttpClient**.
+1. [Download](https://github.com/aspnet/Docs/tree/master/aspnet/web-api/overview/advanced/calling-a-web-api-from-a-net-client/samples/server) and run the server app. [Download instructions](xref:tutorials/index#how-to-download-a-sample). Verify the server app is working. For exaxmple,  `http://localhost:64195/api/products` should return a list of products.
```

Comment:
[Download instructions](xref:tutorials/index#how-to-download-a-sample)
For some reason this will not open in Edge. It tries to open it by asking if I want to search the Windows store for an app to do so. Not sure if this is an Edge problem or what because the link actually looks fine to me. 

Aren't we supposed to put the links to download code in the beginning of the article in the prereqs or intros section? I can't say I've seen specific guidance on that but that has been the consistency with the MVC content guidelines. 
If not, the "Download instructions" phrase should be a complete sentence. 

Also: Verify the server app is working -> Verify the server app works.

Reply:
The downloads won't be on master until the PR is merged into master.
Added link to top.